lib/SimpleSAML/Database.php

```diff
@@ -86,28 +87,34 @@ private function __construct(Configuration $config)
             $driverOptions = [PDO::ATTR_PERSISTENT => true];
         }
 
-        // connect to the master
-        $this->dbMaster = $this->connect(
+        // connect to the primary
+        $this->dbPrimary = $this->connect(
             $config->getString('database.dsn'),
             $config->getString('database.username', null),
             $config->getString('database.password', null),
             $driverOptions
         );
 
-        // connect to any configured slaves
-        $slaves = $config->getArray('database.slaves', []);
-        foreach ($slaves as $slave) {
+        // TODO: deprecated racism: the "database.slave" terminology is preserved here for backwards compatibility.
```

Comment:
This comment is maybe a bit harsh. Just "deprecated" will suffice.

Reply:
Searching for "deprecated" brings up lots of hits, so I was trying to tag it usefully if we needed easy way to find or mark similar legacy occurrences later on, particularly if we tackle the blacklist/whitelist version too. I'll agree that "racism" is a bit harsh - I maybe could have used a reference to knodel or similar instead to create a unique search term.